### ISSUE TITLE
Deadlock after second "Tunnel connection timed out"

### ISSUE THREAD

**Issue by uwedisch** [uwedisch/knxmap#2]
KNXmap is called via Bash script (/bin/bash):
`knxmap -t -p 3671 --nat scan <ip-address> 1.1.1 --bus-info --omit-configuration-reads`

The output is always at the end as shown below and then stucking (forever, at least some minutes):

```
<<<<<<<<<< [INCOMING] <<<<<<<<<<
[misc.py:40 -       trace_outgoing() ] [PACKET TRACE] KnxTunnellingAck source: None, port: None
>>>>>>>>>> [OUTGOING] >>>>>>>>>>
0x06 0x10 0x04 0x21 0x00 0x0A 0x04 0x01 
0x08 0x00 
>>>>>>>>>> [OUTGOING] >>>>>>>>>>
[tunnel.py:67 -   connection_timeout() ] Tunnel connection timed out
[misc.py:40 -       trace_outgoing() ] [PACKET TRACE] KnxDisconnectRequest source: 0.0.0.0, port: 0
>>>>>>>>>> [OUTGOING] >>>>>>>>>>
0x06 0x10 0x02 0x09 0x00 0x10 0x01 0x00 
0x08 0x01 0x00 0x00 0x00 0x00 0x00 0x00 
>>>>>>>>>> [OUTGOING] >>>>>>>>>>
[misc.py:40 -       trace_outgoing() ] [PACKET TRACE] KnxConnectionStateRequest source: 0.0.0.0, port: 0
>>>>>>>>>> [OUTGOING] >>>>>>>>>>
0x06 0x10 0x02 0x07 0x00 0x10 0x01 0x00 
0x08 0x01 0x00 0x00 0x00 0x00 0x00 0x00 
>>>>>>>>>> [OUTGOING] >>>>>>>>>>
^C^CTraceback (most recent call last):
  File "/usr/local/lib/python3.8/dist-packages/KNXmap-0.10.0-py3.8.egg/knxmap/main.py", line 237, in main
  File "/usr/lib/python3.8/asyncio/base_events.py", line 603, in run_until_complete
    self.run_forever()
  File "/usr/lib/python3.8/asyncio/base_events.py", line 570, in run_forever
    self._run_once()
  File "/usr/lib/python3.8/asyncio/base_events.py", line 1823, in _run_once
    event_list = self._selector.select(timeout)
  File "/usr/lib/python3.8/selectors.py", line 468, in select
    fd_event_list = self._selector.poll(timeout, max_ev)
KeyboardInterrupt
```

Same call without Bash script within plain Bash is working. Call is finished in plain Bash after about 45 seconds, within Bash script call was interrupted after about 2 minutes.

**Comment by uwedisch** [comment]
Independent of x64/x86.

**Comment by uwedisch** [comment]
Problem is connected with --bus-info switch.

**Comment by uwedisch** [comment]
Done more tests. There is some problem with NAT mode rsp. routing without NAT. Both have timeout in tunnel connection. Direct connections to the same network without any routing are working fine.

All scan targets are working fine with ETS. So, there is no problem in routing or NAT.

**Comment by uwedisch** [comment]
If the --bus-info switch is not applied on scans for physical addresses it's working until there are no more than a single time out.  With --bus-info switch there are very often two timeouts on non local networks.

So, currently the state is (until bug is fixed):
- Don't use the --bus-info switch on non local networks.
- To reduce the risk of a time out do only scan a single physical address on each scan.

**Comment by uwedisch** [comment]
Same here without NAT mode on local network while scanning huge range of physical addresses, i.e. 1.0.0-15.15.255. So, currently, until the bug is fixed, the state is:

Non local networks:
- Don't use the --bus-info switch.
- To reduce the risk of a time out do only scan a single physical address on each scan.

Local networks:
- To reduce the risk of a time out use the --bus-info switch only with a single physical address on each scan.

**Comment by uwedisch** [comment]
See commit 4ecb498.
Within core.py a call to knxmap.utils.parse_device_descriptor() is made, but never reaching knxmap.utils.parse_device_descriptor().

**Comment by uwedisch** [comment]
Resolved with commit 940a1f5.
